### PR TITLE
improvement(cluster.py): Moved create session from tester.py to cluster.py

### PR DESCRIPTION
**Pull request by Orenef11**
* Trello: https://trello.com/c/MnwSqgfS
   Description:
    ```
     * stop using loader:  https://github.com/scylladb/scylla-cluster-tests/pull/2088
     * move from cqlsh to python driver session
     * add option to filter empty table out: https://github.com/scylladb/scylla-cluster-tests/pull/2088
    ```
* Moving the following functions from tester.py to cluster.py:
  * get_auth_provider
  * _create_session
  * cql_connection
  * cql_connection_exclusive
  * cql_connection_patient
  * cql_connection_patient_exclusive
* Removing test_get_non_system_ks_cf_list.py file
* get_non_system_ks_cf_list:
  * Moving the function from common.py to cluster.py
  * Improvement the function logic (used session inside run_cqlsh function)

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
@Orenef11 
unittests are not passing on this one:
```
10:47:11  __________________ ERROR collecting unit_tests/test_tester.py __________________
10:47:11  ImportError while importing test module '/jenkins/slave/workspace/can_scylla-cluster-tests_PR-2468/unit_tests/test_tester.py'.
10:47:11  Hint: make sure your test modules/packages have valid Python names.
10:47:11  Traceback:
10:47:11  unit_tests/test_tester.py:6: in <module>
10:47:11      from sdcm.tester import ClusterTester, silence, TestResultEvent
10:47:11  sdcm/tester.py:46: in <module>
10:47:11      from sdcm.utils.common import ScyllaCQLSession, get_non_system_ks_cf_list, format_timestamp, \
10:47:11  E   ImportError: cannot import name 'get_non_system_ks_cf_list' from 'sdcm.utils.common' (/jenkins/slave/workspace/can_scylla-cluster-tests_PR-2468/sdcm/utils/common.py)
10:47:11  !!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
10:47:11  =========================== 1 error in 2.60 seconds ============================
```

File: performance_regression_row_level_repair_test.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def _pre_create_schema_scylla_bench(self):
         node = self.db_cluster.nodes[0]
         create_table_query = create_scylla_bench_table_query()
         # pylint: disable=no-member
-        with self.cql_connection_patient(node) as session:
+        with self.db_clustercql_connection_patient(node) as session:
```
Comment (fruch) [inline performance_regression_row_level_repair_test.py:138]:
```suggestion
        with self.db_cluster.cql_connection_patient(node) as session:
```

File: unit_tests/test_get_non_system_ks_cf_list.py [hunk 2]
```diff
@@ -1,134 +0,0 @@
-from __future__ import absolute_import
```
Comment (fruch) [inline unit_tests/test_get_non_system_ks_cf_list.py:1]: where are the tests replace those test ?, or you think they are not needed ?

File: performance_regression_row_level_repair_test.py (hunk 1 above)
Comment (Orenef11) [reply performance_regression_row_level_repair_test.py:138]: Done

> @Orenef11
> unittests are not passing on this one:
> 
> ```
> 10:47:11  __________________ ERROR collecting unit_tests/test_tester.py __________________
> 10:47:11  ImportError while importing test module '/jenkins/slave/workspace/can_scylla-cluster-tests_PR-2468/unit_tests/test_tester.py'.
> 10:47:11  Hint: make sure your test modules/packages have valid Python names.
> 10:47:11  Traceback:
> 10:47:11  unit_tests/test_tester.py:6: in <module>
> 10:47:11      from sdcm.tester import ClusterTester, silence, TestResultEvent
> 10:47:11  sdcm/tester.py:46: in <module>
> 10:47:11      from sdcm.utils.common import ScyllaCQLSession, get_non_system_ks_cf_list, format_timestamp, \
> 10:47:11  E   ImportError: cannot import name 'get_non_system_ks_cf_list' from 'sdcm.utils.common' (/jenkins/slave/workspace/can_scylla-cluster-tests_PR-2468/sdcm/utils/common.py)
> 10:47:11  !!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
> 10:47:11  =========================== 1 error in 2.60 seconds ============================
> ```

thanks!

File: unit_tests/test_get_non_system_ks_cf_list.py (hunk 2 above)
Comment (Orenef11) [reply unit_tests/test_get_non_system_ks_cf_list.py:1]: To use the session I need a real cluster.
This entire file contained tests which are based on factual nodes

**Comment by Orenef11** [comment]
@fruch 
All issues are resolved

**Comment by fruch** [comment]
@Orenef11, 
unit test are failing (I'm guessing it's related to the rebase)
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-2468/7/execution/node/32/log/

**Comment by Orenef11** [comment]
> @Orenef11,
> unit test are failing (I'm guessing it's related to the rebase)
> https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-2468/7/execution/node/32/log/

FIxed

**Comment by bentsi** [comment]
where are backport labels?

**Comment by bentsi** [comment]
why don't you follow PR pre-checks?

File: upgrade_test.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def inner(self, *args, **kwargs):
             base_version = self.params.get('scylla_version', default='')
             system_truncated = bool(parse_version(base_version) >= parse_version('3.1')
                                     and not is_enterprise(base_version))
-            with self.cql_connection_patient(node, keyspace='truncate_ks') as session:
+            with self.db_cluster.cql_connection_patient(node, keyspace='truncate_ks') as session:
```
[the remark this answers is not in the document]
Comment (fruch) [reply upgrade_test.py:40]: we did this change a long time ago, it didn't change the behavior of those functions, it's still get node or nodes as parameter

[the remark this answers is not in the document]
Comment (fruch) [reply upgrade_test.py:40]: Also this specific function would use one node address to initial connect, but the session would connect to all nodes.
`cql_connection_patient_exclusive` needs to be use to communicate only with one node.

**Pull request merged**